### PR TITLE
feat: `lake env` w/o configuration + more

### PR DESCRIPTION
**Pull request by tydeu**
Closes leanprover/lake#179.

Improves `lake env` in a number of ways:
* Allows it to be run without a `lakefile.lean` configuration
* Runing it without a command will display the environment variables set in NAME=VALUE form a la POSIX `env`
* Now adds Lean/Lake and worskapce `bin` directories to `PATH`
* Adds Lean's shared library directories to the shared library path 
* Adds a test

### CODE REVIEW

File: src/lake/Lake/CLI/Help.lean [hunk 1]
```diff
@@ -36,8 +36,8 @@ COMMANDS:
   scripts               shorthand for `lake script list`
   run <script>          shorthand for `lake script run`
   serve                 start the Lean language server
-  env <cmd> [<args>...] execute a command in the workspace's environment
-  exe <exe> [<args>...] build an exe and run it in the workspace's environment
+  env <cmd> [<args>...] execute a command in Lake's environment
```
Comment (semorrison) [inline src/lake/Lake/CLI/Help.lean:39]:
```suggestion
  env                   print the value of environment variables that would be set by `env <cmd>`
  env <cmd> [<args>...] execute a command in Lake's environment
```

Comment (semorrison) [reply src/lake/Lake/CLI/Help.lean:39]: Oh, scratch that, it is in the detailed help.

File: src/lake/Lake/CLI/Help.lean [hunk 2]
```diff
@@ -188,38 +188,42 @@ with the package configuration's `moreServerArgs` field and `args`.
...
-  LD_LIBRARY_PATH       adds the workspace's library directories (other Unix)"
+  LEAN_CC               set to the detected `cc` (if not using the bundled one)
+  LEAN_PATH             adds Lake's and the workspace's Lean library dirs
+  LEAN_SRC_PATH         adds Lake's and the workspace's source dirs
+  PATH                  adds Lean's, Lake's, and the workspace's binary dirs
+  PATH                  adds Lean's and the workspace's library dirs (Windows)
+  DYLD_LIBRARY_PATH     adds Lean's and the workspace's library dirs (MacOS)
+  LD_LIBRARY_PATH       adds Lean's and the workspace's library dirs (other)
+
+If run with no command, Lake will print out the variables set and their values,
```
Comment (semorrison) [inline src/lake/Lake/CLI/Help.lean:214]:
```suggestion
If run with no command, `lake env` will print out the variables set and their values,
```

File: src/lake/test/env/test.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+$LAKE env printenv LAKE_HOME
+$LAKE env printenv LEAN_SYSROOT
+$LAKE env printenv LEAN_AR | grep -m1 ar
+$LAKE env printenv LEAN_PATH
+$LAKE -d ../../examples/hello env printenv LEAN_PATH | grep -m1 examples/hello
+$LAKE env printenv LEAN_SRC_PATH | grep -m1 lake
+$LAKE -d ../../examples/hello env printenv LEAN_SRC_PATH | grep -m1 examples/hello
+$LAKE -d ../../examples/hello env printenv PATH | grep -m1 examples/hello
+
+# Test that `env` perserves environment for certain variables
```
Comment (semorrison) [inline src/lake/test/env/test.sh:23]:
```suggestion
# Test that `env` preserves environment for certain variables
```

File: src/lake/test/env/test.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+LAKE=${LAKE:-../../build/bin/lake}
+
+# Test the functionality of `lake env`
+# Also test https://github.com/leanprover/lake/issues/179
+
+# Test `env` with no command
+$LAKE env
+
+# Test installation variables are set
```
Comment (semorrison) [inline src/lake/test/env/test.sh:12]:
```suggestion
# Test installation variables are set
# Note: `set -euxo pipefail` above ensures these fail if the variable is not set.
```
Maybe not necessary, but it took me a moment to understand how this was actually testing anything.

File: src/lake/test/env/test.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+LAKE=${LAKE:-../../build/bin/lake}
+
+# Test the functionality of `lake env`
+# Also test https://github.com/leanprover/lake/issues/179
+
+# Test `env` with no command
+$LAKE env
```
Comment (semorrison) [inline src/lake/test/env/test.sh:10]: Test this actually returns something? Just ` | grep ...` would do.

**Comment by tydeu** [comment]
@Kha, you removed your review request a while back. Should I assume that means this is fine to merge as-is?

**Comment by Kha** [comment]
It means that if you (and the other reviewers) are happy with it, I'm happy with it :) . I didn't review it.

**Pull request merged**